mlflow/spark.py

```diff
@@ -501,7 +508,14 @@ def save_model(
     tmp_path = _tmp_path(dfs_tmpdir)
     spark_model.save(tmp_path)
     sparkml_data_path = os.path.abspath(os.path.join(path, _SPARK_MODEL_PATH_SUB))
-    _HadoopFileSystem.copy_to_local_file(tmp_path, sparkml_data_path, remove_src=True)
+    # If spark DFS is DBFS and we're running on a Databricks cluster, copy to local FS
+    # via the FUSE mount
+    is_saving_to_dbfs = is_valid_dbfs_uri(tmp_path) or posixpath.abspath(tmp_path) == tmp_path
+    if is_saving_to_dbfs and databricks_utils.is_in_cluster():
+        tmp_path_fuse = dbfs_hdfs_uri_to_fuse_path(tmp_path)
+        shutil.move(src=tmp_path_fuse, dst=sparkml_data_path)
```

Comment:
When does the file get written to tmp_path fuse so we can move it? Are we assuming that the FUSE mount has been synced with HDFS? What if the sync hasn't finished yet?

Reply:
In this save_model method, the goal is to save the SparkML model to a local filesystem path

The flow here is:
* Save SparkML model to a temporary dbfs location (`dbfs:/tmp/my-model`)
* Move that file to the local filesystem using FUSE and a file move (move `/dbs/tmp/my-model` to the `/my/desired/local/path`)

We leverage FUSE to copy from DBFS -> local as there's no allowlisted JVM-side, HDFS API for copying from dbfs:/... to the local filesystem

Reply:
The FUSE access has the same consistency properties as standard HDFS-style access to the same DBFS path. That is, writing to `dbfs:/tmp/my-model` and reading `/dbfs/tmp/my-model` is equivalent from a blob-storage-eventual-consistency perspective to writing to `dbfs:/tmp/my-model` and subsequently reading `dbfs:/tmp/my-model`